R/read_habitatdata.R

```diff
@@ -691,69 +714,76 @@ read_watersurfaces <-
       watersurfaces %>%
       {
         if (version == "watersurfaces_v1.2") {
-          rename(., water_level_management = .data$PEILBEHEER)
+          rename(., water_level_management = "PEILBEHEER",
+                 hyla_code = "HYLAC")
+        } else if (version == "watersurfaces_v2024") {
+          rename(.,
+                 wfd_type_alternative = "KRWTYPEA",
+                 water_level_management = "PEILBEHEER") %>%
+            mutate(
+              across(where(is.character), ~na_if(., "")))
         } else {
-          .
+          rename(., hyla_code = "HYLAC")
         }
       } %>%
       select(
-        polygon_id = .data$WVLC,
-        wfd_code = .data$WTRLICHC,
-        hyla_code = .data$HYLAC,
-        name = .data$NAAM,
-        area_name = .data$GEBIED,
-        wfd_type = .data$KRWTYPE,
-        wfd_type_certain = .data$KRWTYPES,
-        depth_class = .data$DIEPKL,
-        connectivity = .data$CONNECT,
-        usage = .data$FUNCTIE,
-        matches("^water_level_management$")
+        polygon_id = "WVLC",
+        wfd_code = "WTRLICHC",
+        any_of("hyla_code"),
+        name = "NAAM",
+        area_name = "GEBIED",
+        wfd_type = "KRWTYPE",
+        any_of("wfd_type_alternative"),
+        wfd_type_certain = "KRWTYPES",
+        depth_class = "DIEPKL",
+        connectivity = "CONNECT",
+        usage = "FUNCTIE",
+        any_of("water_level_management")
       ) %>%
-      mutate(depth_class = str_replace(
-        string = .data$depth_class,
-        pattern = "\u2265",
-        replacement = ">="
-      )) %>%
-      mutate(across(
-        c(
-          .data$area_name,
-          .data$depth_class,
-          .data$connectivity,
-          .data$usage,
-          matches("^water_level_management$")
-        ),
-        as.factor
-      )) %>%
       mutate(
+        depth_class = str_replace(
+          string = .data$depth_class,
+          pattern = "\u2265",
+          replacement = ">="),
+        across(
+          c(
+            "area_name",
+            "depth_class",
+            "connectivity",
+            "usage",
+            any_of("water_level_management")
+          ),
+          as.factor),
         wfd_type = .data$wfd_type %>%
           factor(
             levels =
               levels(wfd_typetransl$wfd_type)
-          ),
-        hyla_code = ifelse(.data$hyla_code == 0,
-          NA,
-          .data$hyla_code
-        )
-      ) %>%
-      arrange(.data$polygon_id)
+            ),
+        across(
+          any_of("wfd_type_alternative"),
+          ~ factor(.,
+                   levels =
+                     levels(wfd_type_alttransl$wfd_type_alternative)
+                   )),
+        across(
+          any_of("hyla_code"),
+          ~ ifelse(.data$hyla_code == 0,
+                   NA,
+                   .data$hyla_code))
```

Comment:
The function passed to `across()` does not really use the column selected by `across()` right now, but reselects it from `.data`.

So I believe it's better like this:

```suggestion
        across(
          any_of("hyla_code"),
          ~ ifelse(.x == 0,
                   NA,
                   .x))
```

Reply:
done in https://github.com/inbo/n2khab/pull/192/commits/fe005033308d91c16572c2ab322b45391ebfad8c